qiskit/test/mock/fake_backend.py

```diff
@@ -38,6 +62,257 @@ def __init__(self, token="123456", url="https://"):
+
+            if pulse_job:
+                from qiskit.providers.aer.pulse import PulseSystemModel
+
+                system_model = PulseSystemModel.from_backend(self)
+                sim = aer.Aer.get_backend("pulse_simulator")
+                job = sim.run(circuits, system_model=system_model, **kwargs)
+            else:
+                sim = aer.Aer.get_backend("qasm_simulator")
+                if self._props_dict:
+                    noise_model = self._get_noise_model_from_backend_v2(self, warnings=False)
+                    job = sim.run(circuits, noise_model=noise_model, **kwargs)
+                else:
+                    job = sim.run(circuits, **kwargs)
+        else:
+            if pulse_job:
+                raise QiskitError("Unable to run pulse schedules without qiskit-aer installed")
+            warnings.warn("Aer not found using BasicAer and no noise", RuntimeWarning)
+            sim = basicaer.BasicAer.get_backend("qasm_simulator")
+            job = sim.run(circuits, **kwargs)
+        return job
+
+    def _get_noise_model_from_backend_v2(
+        self,
+        gate_error=True,
+        readout_error=True,
+        thermal_relaxation=True,
+        temperature=0,
+        gate_lengths=None,
+        gate_length_units="ns",
+        standard_gates=None,
+        warnings=True,
+    ):
+        """Build noise model from BackendV2.
+        This is a temporary fix until Aer supports V2 backends.
+        """
+
+        from warnings import warn, catch_warnings, filterwarnings
+        from qiskit.circuit import Delay
+        from qiskit.providers.exceptions import BackendPropertyError
+        from qiskit.providers.aer.noise import NoiseModel
+        from qiskit.providers.aer.noise.device.models import (
+            _excited_population,
+            basic_device_gate_errors,
+            basic_device_readout_errors,
+        )
+        from qiskit.providers.aer.noise.passes import RelaxationNoisePass
+
+        properties = BackendProperties.from_dict(self._props_dict)
+        basis_gates = self.operation_names
+        num_qubits = self.num_qubits
+        dt = self.dt
+
+        noise_model = NoiseModel(basis_gates=basis_gates)
+
+        # Add single-qubit readout errors
+        if readout_error:
+            for qubits, error in basic_device_readout_errors(properties):
+                noise_model.add_readout_error(error, qubits, warnings=warnings)
+
+        if standard_gates is not None:
+            warn(
+                '"standard_gates" option has been deprecated as of qiskit-aer 0.10.0'
+                " and will be removed no earlier than 3 months from that release date.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+        # Add gate errors
+        with catch_warnings():
+            filterwarnings(
+                "ignore",
+                category=DeprecationWarning,
+                module="qiskit.providers.aer.noise.device.models",
+            )
```

Comment:
What is this for?

Reply:
Same as above. I am not sure about this. I literally just copied from Aer https://github.com/Qiskit/qiskit-aer/blob/71d55588a338c02740e857fceef8439412746e20/qiskit/providers/aer/noise/noise_model.py#L348-L353

Reply:
Took a closer look at this. It was introduced rather recently in the refactoring PR: https://github.com/Qiskit/qiskit-aer/pull/1141 @itoko What is the `filterwarnings` for? To give you some context, we are adding some part of Aer code `NoiseModel.from_backend` to Terra as a temporary fix to support noisy simulation of qasm jobs in mocked backends.